src/mainwindow.cpp

```diff
@@ -752,6 +755,31 @@ void MainWindow::find()
     consoleTabulator->terminalHolder()->currentTerminal()->impl()->toggleShowSearchBar();
 }
 
+void MainWindow::handleHistory()
+{
+    const QString dir = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
+    QDir().mkpath(dir);
+    const QString fn = dir + QLatin1String("/qterminal.history.") + QString::number(QCoreApplication::applicationPid());
+    QFile file(fn);
+    if (!file.open(QIODevice::WriteOnly)) {
+        qDebug() << "Failed to open" << file.fileName() << "for writing";
+        return;
+    }
+    TermWidgetImpl *impl = consoleTabulator->terminalHolder()->currentTerminal()->impl();
+    impl->saveHistory(&file);
+    file.close();
+    QStringList args = Properties::Instance()->handleHistoryCommand.split(QLatin1Char(' '), QString::SkipEmptyParts);
+    if (args.isEmpty())
+        return;
```

Comment:
It should be better to have a message If handleHistoryCommand is not set, so that users who want to open files manually can get the filename.